### ISSUE TITLE
Pixel / Pixel XL support

### ISSUE THREAD

**Issue by sorccu** [openstf/stf#441]
Now that the [Pixel phone](https://madeby.google.com/phone/) has been released, we should verify that it works with STF.

Unfortunately, it's unclear when orders will open in Japan where we're located, in the past it has taken quite a while, even months. Even if you try to purchase using VPN from the US store, it won't let you select a non-US payment method. Although likely possible with enough time and effort, it's not worth it to attempt to work around all the restrictions.

In short, it's unclear if we'll actually have a device before real users do.

On the off chance that someone wants to donate one of these devices (w/ any specs), please feel free to offer.

Issues:

- [ ] Android 7.1 support (not available yet)

**Comment by thechirinos** [comment]
While I can't donate my Pixel, I wanted to share the errors I'm seeing when STF is "Preparing" the device:
```
INF/provider 82174 [*] Cleaning up device worker "FA69T0304880"
ERR/provider 82174 [*] Device worker "FA69T0304880" died with code 1
INF/provider 82174 [*] Restarting device worker "FA69T0304880"



INF/device:support:push 82197 [FA69T0304880] Sending output to "tcp://127.0.0.1:7116"
INF/device 82197 [FA69T0304880] Preparing device

INF/device:support:sub 82197 [FA69T0304880] Receiving input from "tcp://127.0.0.1:7114"
INF/device:support:sub 82197 [FA69T0304880] Subscribing to permanent channel "*ALL"
INF/device:support:properties 82197 [FA69T0304880] Loading properties
INF/device:support:abi 82197 [FA69T0304880] Supports ABIs arm64-v8a, armeabi-v7a, armeabi
FTL/device 82197 [FA69T0304880] Setup had an error Error: At least one of these paths should exist: /usr/local/lib/node_modules/stf/vendor/minicap/shared/android-7.1/arm64-v8a/minicap.so, /usr/local/lib/node_modules/stf/vendor/minicap/shared/android-25/arm64-v8a/minicap.so, /usr/local/lib/node_modules/stf/vendor/minicap/shared/android-7.1/armeabi-v7a/minicap.so, /usr/local/lib/node_modules/stf/vendor/minicap/shared/android-25/armeabi-v7a/minicap.so, /usr/local/lib/node_modules/stf/vendor/minicap/shared/android-7.1/armeabi/minicap.so, /usr/local/lib/node_modules/stf/vendor/minicap/shared/android-25/armeabi/minicap.so
    at Object.module.exports.requiredMatch (/usr/local/lib/node_modules/stf/lib/util/pathutil.js:33:9)
    at /usr/local/lib/node_modules/stf/lib/units/device/resources/minicap.js:35:23
    at SerialSyrup.ParallelSyrup.invoke (/usr/local/lib/node_modules/stf/node_modules/stf-syrup/lib/parallel.js:54:24)
    at /usr/local/lib/node_modules/stf/node_modules/stf-syrup/lib/serial.js:43:33
    at tryCatch1 (/usr/local/lib/node_modules/stf/node_modules/stf-syrup/node_modules/bluebird/js/main/util.js:63:19)
    at Promise$_callHandler [as _callHandler] (/usr/local/lib/node_modules/stf/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:708:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/usr/local/lib/node_modules/stf/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:724:18)
    at Promise$_settlePromiseAt [as _settlePromiseAt] (/usr/local/lib/node_modules/stf/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:896:14)
    at Promise$_fulfillPromises [as _fulfillPromises] (/usr/local/lib/node_modules/stf/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:1041:14)
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/usr/local/lib/node_modules/stf/node_modules/stf-syrup/node_modules/bluebird/js/main/async.js:74:12)
    at Async$consumeFunctionBuffer (/usr/local/lib/node_modules/stf/node_modules/stf-syrup/node_modules/bluebird/js/main/async.js:37:14)
    at nextTickCallbackWith0Args (node.js:415:9)
    at process._tickCallback (node.js:344:13)
FTL/util:lifecycle 82197 [FA69T0304880] Shutting down due to fatal error
INF/provider 82174 [*] Cleaning up device worker "FA69T0304880"
ERR/provider 82174 [*] Device worker "FA69T0304880" died with code 1
INF/provider 82174 [*] Restarting device worker "FA69T0304880"

INF/device:support:push 82198 [FA69T0304880] Sending output to "tcp://127.0.0.1:7116"
INF/device 82198 [FA69T0304880] Preparing device
```

It appears to continually loop through these errors-- I saw them at least 40 times. 

Hope this helps!

**Comment by sorccu** [comment]
I think you're using an old version. Be sure to use STF 2.1.0 or newer. Would be great to hear if it works!

**Comment by thechirinos** [comment]
You're 100% correct. I was on an older version. My Pixel is working great on it now. Thanks!

**Comment by sorccu** [comment]
Great! One more thing, do you see the following message in the log output?

> Found 7.1 Developer Preview SurfaceComposerClient::setDisplaySurface

Should be grouped under screen or minicap.

**Comment by thechirinos** [comment]
I'm not seeing it. Here's what it looks like when I plug in the Pixel:
```
INF/provider 91249 [*] Found device "FA69V0309310" (offline)
INF/reaper 91248 [reaper001] Device "FA69V0309310" is present
INF/provider 91249 [*] Registered device "FA69V0309310"
INF/provider 91249 [*] Device "FA69V0309310" is now "device" (was "offline")
INF/device:support:push 91312 [FA69V0309310] Sending output to "tcp://127.0.0.1:7116"
INF/device 91312 [FA69V0309310] Preparing device
INF/device:support:sub 91312 [FA69V0309310] Receiving input from "tcp://127.0.0.1:7114"
INF/device:support:sub 91312 [FA69V0309310] Subscribing to permanent channel "*ALL"
INF/device:support:properties 91312 [FA69V0309310] Loading properties
INF/device:support:abi 91312 [FA69V0309310] Supports ABIs arm64-v8a, armeabi-v7a, armeabi
INF/device:resources:service 91312 [FA69V0309310] Checking whether we need to install STFService
INF/device:resources:service 91312 [FA69V0309310] Running version check
INF/device:resources:service 91312 [FA69V0309310] STFService up to date
INF/device:plugins:service 91312 [FA69V0309310] Launching agent
INF/device:plugins:service 91312 [FA69V0309310] Agent says: "Listening on @stfagent"
INF/device:plugins:service 91312 [FA69V0309310] Launching service
INF/device:plugins:service 91312 [FA69V0309310] Agent says: "InputClient started"
INF/device:plugins:display 91312 [FA69V0309310] Reading display info
INF/device:plugins:phone 91312 [FA69V0309310] Fetching phone info
INF/device:plugins:identity 91312 [FA69V0309310] Solving identity
INF/device:plugins:solo 91312 [FA69V0309310] Subscribing to permanent channel "ChWTiudFEoUAM9LeyywnbA4zp64="
INF/device:plugins:screen:stream 91312 [FA69V0309310] Starting WebSocket server on port 7412
WRN/device:plugins:data 91312 [FA69V0309310] Unable to find device data { serial: 'FA69V0309310',
  platform: 'Android',
  manufacturer: 'GOOGLE',
  operator: null,
  model: 'Pixel',
  version: '7.1',
  abi: 'arm64-v8a',
  sdk: '25',
  product: 'sailfish',
  display: 
   { id: 0,
     width: 1080,
     height: 1920,
     xdpi: 442.45098876953125,
     ydpi: 443.3450012207031,
     fps: 60.000003814697266,
     density: 2.625,
     rotation: 0,
     secure: true,
     size: 4.971247790102727,
     url: 'ws://localhost:7412' },
  phone: { imei: '352689080865706', network: 'UNKNOWN' } }
INF/device:plugins:touch 91312 [FA69V0309310] Touch origin is top left
INF/device:plugins:touch 91312 [FA69V0309310] Requesting touch consumer to start
INF/device:plugins:touch 91312 [FA69V0309310] Launching screen service
INF/device:plugins:touch 91312 [FA69V0309310] Connecting to minitouch service
INF/device:plugins:touch 91312 [FA69V0309310] minitouch says: "Note: device /dev/input/mouse1 is not supported by libevdev"
INF/device:plugins:touch 91312 [FA69V0309310] minitouch says: "Note: device /dev/input/mouse0 is not supported by libevdev"
INF/device:plugins:touch 91312 [FA69V0309310] minitouch says: "Note: device /dev/input/mice is not supported by libevdev"
INF/device:plugins:touch 91312 [FA69V0309310] minitouch says: "Type B touch device synaptics_dsxv26 (1079x1919 with 10 contacts) detected on /dev/input/event3 (score 2100)"
INF/device:plugins:touch 91312 [FA69V0309310] Reading minitouch banner
INF/device:plugins:touch 91312 [FA69V0309310] minitouch says: "Connection established"
INF/device:plugins:vnc 91312 [FA69V0309310] Starting VNC server on port 7414
INF/device:plugins:browser 91312 [FA69V0309310] Loading browser list
INF/device:plugins:browser 91312 [FA69V0309310] Updating browser list
INF/device:plugins:mute 91312 [FA69V0309310] Will not mute master volume during use
INF/device:plugins:forward 91312 [FA69V0309310] Launching reverse port forwarding service
INF/device:plugins:forward 91312 [FA69V0309310] Connecting to reverse port forwarding service
INF/device:plugins:connect 91312 [FA69V0309310] Listening on port 7413
INF/device 91312 [FA69V0309310] Fully operational
INF/util:datautil 91253 [*] Device database does not have an entry for "FA69V0309310" (model "Pixel"/"sailfish"). This is not an error, it only means that we don't have data for it yet.
INF/provider 91249 [*] Providing all 2 device(s)
```

This is what it shows when I take control of the device:
```
INF/util:datautil 91252 [*] Device database does not have an entry for "FA69V0309310" (model "Pixel"/"sailfish"). This is not an error, it only means that we don't have data for it yet.
IMP/device:plugins:group 91312 [FA69V0309310] Now owned by "test@email.com"
INF/device:plugins:group 91312 [FA69V0309310] Subscribing to group channel "VjMVDeIXSn6rfp30O0nr+Q=="
INF/util:datautil 91252 [*] Device database does not have an entry for "FA69V0309310" (model "Pixel"/"sailfish"). This is not an error, it only means that we don't have data for it yet.
INF/device:plugins:screen:stream 91312 [FA69V0309310] Keeping 801x1328 as current frame producer projection
INF/device:plugins:screen:stream 91312 [FA69V0309310] Requesting frame producer to start
INF/device:plugins:screen:stream 91312 [FA69V0309310] Launching screen service
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "PID: 18167"
INF/device:plugins:screen:stream 91312 [FA69V0309310] Connecting to minicap service
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "INFO: Using projection 1080x1920@747x1328/0"
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "INFO: (external/MY_minicap/src/minicap_25.cpp:241) Creating SurfaceComposerClient"
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "INFO: (external/MY_minicap/src/minicap_25.cpp:244) Performing SurfaceComposerClient init check"
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "INFO: (external/MY_minicap/src/minicap_25.cpp:251) Creating virtual display"
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "INFO: (external/MY_minicap/src/minicap_25.cpp:257) Creating buffer queue"
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "INFO: (external/MY_minicap/src/minicap_25.cpp:262) Creating CPU consumer"
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "INFO: (external/MY_minicap/src/minicap_25.cpp:266) Creating frame waiter"
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "INFO: (external/MY_minicap/src/minicap_25.cpp:270) Publishing virtual display"
IMP/device:plugins:connect 91312 [FA69V0309310] Remote Connect Started for device "FA69V0309310" at "localhost:7413"
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "INFO: (jni/minicap/JpgEncoder.cpp:64) Allocating 6268932 bytes for JPG encoder"
INF/device:plugins:screen:stream 91312 [FA69V0309310] Reading minicap banner
INF/device:plugins:screen:stream 91312 [FA69V0309310] minicap says: "INFO: (jni/minicap/minicap.cpp:444) New client connection"
```

**Comment by sorccu** [comment]
Thanks, awesome info! Means that unlike the 7.1 Developer Preview build, they're actually using the AOSP source they published.
